app/include/zmk/hid.h

```diff
@@ -164,8 +164,8 @@ struct zmk_hid_consumer_report {
 
 int zmk_hid_register_mod(zmk_mod modifier);
 int zmk_hid_unregister_mod(zmk_mod modifier);
-int zmk_hid_register_mods(zmk_mod_flags modifiers);
-int zmk_hid_unregister_mods(zmk_mod_flags modifiers);
+int zmk_hid_implicit_modifiers_press(zmk_mod_flags implicit_modifiers);
+int zmk_hid_implicit_modifiers_release();
```

Comment:
I'm not sure if the modifier implicit/explicit/counting state and related functions should live within `hid.h/c`.

Rationale:
 - it reduces the cohesion of the `hid` module.
 - there's weak coupling to the rest of the `hid` module.
 - it feels like a conceptual layer above `hid`.
 - the `modifiers` system will likely be part of or linked to the *internal state* subsystem(s) that have been discussed recently.
 - `hid.c` itself can probably be refactored into `hid_keyboard.c` and `hid_consumer.c`

Perhaps it's better suited to a dedicated module, say `modifiers.h/c` or `hid_modifiers.h/c`?

Reply:
I think this is fine for now, and we can refactor if/when we have a plan/need for the state changes.